File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    
    sort(citations.begin(), citations.end());
        
    int count=0;
    int num = 0;
    int i=0;
    
    while(1){
        count = 0;
        num++;
        for(auto c: citations)
            if(num <= c) count ++;
        if(count < num) break;
    }
    answer = num-1;
    
    return answer;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    
    citations.sort(reverse=True) # 큰 순서대로 정렬
    answer = citations[0] # 최댓값부터 시작
    
    while answer >= 0:
        # answer 이상인 값의 수 >= answer면 return answer
        cnt = 0
        for c in citations:
            if c >= answer:
                cnt += 1
        if cnt >= answer:
            return answer
    
        answer -= 1